### ISSUE TITLE
Removing record on filtered list doesn't refresh list

### ISSUE THREAD

**Issue by pwasiewicz** [MaKleSoft/padlock#176]
When records list is filtered
![not-deleted](https://user-images.githubusercontent.com/8725777/39320983-f8198744-4985-11e8-84e7-bcb791d49367.png)
and we decide to remove record
![del_prompt](https://user-images.githubusercontent.com/8725777/39321002-04990396-4986-11e8-9572-cf35d1ddc606.jpg)
then records is still visible on filtered list (it seems that it is not refreshed after removal)
![not-deleted](https://user-images.githubusercontent.com/8725777/39321035-1acec79a-4986-11e8-8f7f-0a2c7ac3ec82.png)

**Comment by MaKleSoft** [comment]
Tried to reproduce this without success. Is the record actually being removed? I.e. if you restart the app, is it still there?

**Comment by pwasiewicz** [comment]
Yes. record is removed. What is more, I do not need to restart app. It is enough to filter records list with different string (any) and list is refreshed properly.
